Game/world/world.h

```diff
@@ -175,6 +176,19 @@ class World final {
     void                 addFreePoint  (const Tempest::Vec3& pos, const Tempest::Vec3& dir, const char* name);
     void                 addSound      (const ZenLoad::zCVobData& vob);
 
+    inline bool addItemOrNpcBySymbolName (const std::string& name, const Tempest::Vec3& at) {
+      bool ret = false;
+      size_t id = script().getSymbolIndex(name);
+
+      if( addNpc(id, at) != nullptr ) {
```

Comment:
```
if(script().getSymbol(id).instance.instanceOf(Daedalus::IC_Npc))
  addNpc(..)
if(script().getSymbol(id).instance.instanceOf(Daedalus::IC_Item))
  addItem(..)
```

Reply:
This does not work, I tried that. Mosters seem to be IC_None for some reason.

Reply:
Right, `instanceOf` work only on already created instances.
Anyway I think, that try-and-fail is not a great approach to instantiate object by name. 

I take a quick look on Daedalus to see what is possible to do here:
```
  auto& obj = vm.getDATFile().getSymbolByName(name); // get's you a symbol props
  // obj.properties.elemProps.type = EParType_Instance, for instances
  // obj.properties.elemProps.type = EParType_Class, for classes
  auto& cls = vm.getDATFile().getSymbolByIndex(s.parent); // get's you a parent object or prototype
  cls.name=="C_NPC" // for npc
```
So my suggestion is to unroll s.parent chain until parent=size_t(-1), or C_NPC/C_ITEM is found.

Reply:
Yea I also don't like the current implementation, its more a workaround than a implementation.
I will take a look at this again when I have some more time.

I started to look into the code when I started to create this pull request, and for some issues, so I currently do not have much of a understanding of the code especially about the properties which could be used and how everything works together.

Could take some time till I get to it, but thanks for looking into it.